src/aegis_ai_ml/src/classifier/cve-impact-classifier/cve_feature_extraction.py

```diff
@@ -0,0 +1,758 @@
+"""
+Extracts features from scraped CVE data using the logic from the Perl script written by Alexander : 
+https://gitlab.cee.redhat.com/prodsec/psirt/al-kernel/-/blob/main/alkernel_curver_daemon.pl
+WITH DIAGNOSTICS AND TIMEOUT PROTECTION
+"""
+
+import os
+import re
+import json
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from typing import Dict, List, Tuple, Optional
+import logging
+from html import unescape
+import subprocess
+import time
+import signal
+
+# Setup logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+class TimeoutException(Exception):
+    pass
+
+def timeout_handler(signum, frame):
+    raise TimeoutException("CVE processing timeout")
+
+class CVEFeatureExtractor:
+    def __init__(self, data_dir: str = "data"):
+        self.data_dir = Path(data_dir)
+        
+        # Severity mappings from Perl code
+        self.severity_map = {
+            'IMPORTANT': 0,
+            'MODERATE': 1, 
+            'LOW': 2
+        }
+        
+        # Feature names (48 features as in Perl code)
+        self.feature_names = [
+            'kasan', 'lock', 'write', 'read', 'fixprintf', 'remote', 'danger', 'bpf',
+            'nullptr', 'disk', 'init', 'warnonly', 'outofbounds', 'deadlock', 'trace',
+            'usb', 'nomemchk', 'virt', 'simplefix', 'netfilter', 'nvme', 'race',
+            'leak', 'errorpath', 'uaf', 'networking', 'notincludedcomponent', 'skb',
+            'dma', 'logger', 'improve', 'kernel_panic_plus_uaf', 'timer', 'spectre',
+            'hardware', 'servertoclientfail', 'tipc', 'debugfs', 'linus', 'cveoutdated',
+            'boot', 'test', 'qdisc', 'ipv6', 'iouring', 'syzbot', 'kernel_panic', 'memory'
+        ]
+        
+        # Ground truth labels:
+        self.ground_truth_labels = self._load_ground_truth()
+
+    def _load_ground_truth(self) -> Dict[str, str]:
+        """Load ground truth severity labels from data/kernel_cves.json"""
+        labels = {}
+        # Load from kernel_cves.json
+        try:
+            with open("data/kernel_cves.json", 'r') as f:
+                train_data = json.load(f)
+            for cve in train_data:
+                labels[cve['cve_id']] = cve['impact']
+            print(f"Loaded {len(train_data)} CVEs with labels")
+        except FileNotFoundError:
+            print("kernel_cves.json not found")
+
+        return labels
+    def clean_content(self, content: str) -> str:
+        """Clean and normalize content"""
+        # limiting size for mem 
+        if len(content) > 10 * 1024 * 1024:  # 10MB limit
```

Comment:
This duplicates the size check in `extract_patch_features()` but with higher threshold.  So the `if` body is unreachable.